Spell fixes and md update

SahilKr24 opened the pull request:
Signed-off-by: Sahil <sahil@chaosnative.com>

1. Added missing &apos; and commas.
2. `a v̶a̶l̶u̶a̶b̶e̶ resource` > `a valuable resource` on `/mentor/building-a-lifetime-contributor`

rspier commented:
I'm sorry, I thought I replied to this.  Apparently I never hit `Comment`.

Please split the content changes and the formatting changes into separate PRs.

I'm also not sure we want the all the formatting changes, Markdown is a flexible language.  Changing all the lists from `-` to `+` isn't necessary.  I setup markdownlint a few weeks ago, which should help with the most egregious issues.

SahilKr24 commented:
> I'm sorry, I thought I replied to this. Apparently I never hit `Comment`.
> 
> Please split the content changes and the formatting changes into separate PRs.
> 
> I'm also not sure we want the all the formatting changes, Markdown is a flexible language. Changing all the lists from `-` to `+` isn't necessary. I setup markdownlint a few weeks ago, which should help with the most egregious issues.

Thanks for the review, I have updated the PR accordingly. The markdown issue was mostly due to Prettier, I fixed it. Please let me know if any other changes are required.

rspier commented:
Thank you for the fixes.  I'm amazed that people keep finding things after all these years.
